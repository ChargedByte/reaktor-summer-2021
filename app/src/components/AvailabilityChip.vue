<template>
  <v-chip :color="color" small>{{ text }}</v-chip>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component
export default class AvailabilityChip extends Vue {
  @Prop()
  value!: string

  get color(): string {
    switch (this.value) {
      case 'InStock':
        return 'success'
      case 'LessThan10':
        return 'warning'
      case 'OutOfStock':
        return 'error'
      default:
        return 'grey'
    }
  }

  get text(): string {
    switch (this.value) {
      case 'InStock':
        return 'In Stock'
      case 'LessThan10':
        return 'Less than 10'
      case 'OutOfStock':
        return 'Out of Stock'
      default:
        return 'Unknown'
    }
  }
}
</script>
