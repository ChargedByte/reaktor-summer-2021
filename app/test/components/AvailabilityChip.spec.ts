import { createLocalVue, mount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import AvailabilityChip from '~/components/AvailabilityChip.vue'

describe('AvailabilityChip', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(AvailabilityChip, { localVue, vuetify })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly: empty', () => {
    const wrapper = mount(AvailabilityChip, { localVue, vuetify })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: InStock', () => {
    const wrapper = mount(AvailabilityChip, {
      localVue,
      vuetify,
      propsData: { value: 'InStock' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: LessThan10', () => {
    const wrapper = mount(AvailabilityChip, {
      localVue,
      vuetify,
      propsData: { value: 'LessThan10' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: OutOfStock', () => {
    const wrapper = mount(AvailabilityChip, {
      localVue,
      vuetify,
      propsData: { value: 'OutOfStock' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
