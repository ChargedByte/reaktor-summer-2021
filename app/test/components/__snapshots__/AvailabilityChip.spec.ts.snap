// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityChip renders correctly: InStock 1`] = `"<span class=\\"v-chip theme--light v-size--small success\\"><span class=\\"v-chip__content\\">In Stock</span></span>"`;

exports[`AvailabilityChip renders correctly: LessThan10 1`] = `"<span class=\\"v-chip theme--light v-size--small warning\\"><span class=\\"v-chip__content\\">Less than 10</span></span>"`;

exports[`AvailabilityChip renders correctly: OutOfStock 1`] = `"<span class=\\"v-chip theme--light v-size--small error\\"><span class=\\"v-chip__content\\">Out of Stock</span></span>"`;

exports[`AvailabilityChip renders correctly: empty 1`] = `"<span class=\\"v-chip theme--light v-size--small grey\\"><span class=\\"v-chip__content\\">Unknown</span></span>"`;
